pkg/autoscaler/scaling/autoscaler_test.go

```diff
@@ -142,12 +149,15 @@ func TestAutoscalerMetricsWithRPS(t *testing.T) {
 	spec := a.currentSpec()
 
 	expectScale(t, a, time.Now().Add(61*time.Second), ScaleResult{10, ebc, na, true})
-	metricstest.CheckLastValueData(t, stableRPSM.Name(), wantTags, 100)
-	metricstest.CheckLastValueData(t, panicRPSM.Name(), wantTags, 100)
-	metricstest.CheckLastValueData(t, desiredPodCountM.Name(), wantTags, 10)
-	metricstest.CheckLastValueData(t, targetRPSM.Name(), wantTags, spec.TargetValue)
-	metricstest.CheckLastValueData(t, excessBurstCapacityM.Name(), wantTags, float64(ebc))
-	metricstest.CheckLastValueData(t, panicM.Name(), wantTags, 1)
+	wantMetrics := []metricstest.Metric{
+		metricstest.FloatMetric(stableRPSM.Name(), 100, wantTags).WithResource(wantResource),
+		metricstest.FloatMetric(panicRPSM.Name(), 100, wantTags).WithResource(wantResource),
+		metricstest.IntMetric(desiredPodCountM.Name(), 10, wantTags).WithResource(wantResource),
+		metricstest.FloatMetric(targetRPSM.Name(), spec.TargetValue, wantTags).WithResource(wantResource),
+		metricstest.FloatMetric(excessBurstCapacityM.Name(), float64(ebc), wantTags).WithResource(wantResource),
+		metricstest.IntMetric(panicM.Name(), 1, wantTags).WithResource(wantResource),
```

Comment:
Given that `wantTags` is empty, I wonder if it would make sense for `FooMetrics` to have the tag map be optional... (Can do this after this PR.)

Reply:
Alternatively, you can probably pass `nil` here.

Reply:
Yes,  `nil` works.